pkg/authorization/resolver_test.go

```diff
@@ -128,16 +129,19 @@ func TestResolverWithWarrants(t *testing.T) {
 			wantResourceRules:    []authorizer.ResourceRuleInfo{getServices},
 			wantNonResourceRules: nil, // global service accounts do no work without a cluster.
 		},
+		// TODO(cnvergence): restore the commented lines once we drop the global service account feature flag
```

Comment:
couldn't we just make this dependent on the feature gate? So the test is active again when the gate is on or the code here breaks when the gate goes away?

Reply:
like add a `skip bool` field and make that dependent on the gate.

Reply:
That also makes a perfect sense